src/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleTransform.java

```diff
@@ -75,33 +77,50 @@ public Rule applyRestToApp(RestRule rest, Rule rule, User user) {
         return groups;
     }
 
-    public RestRule appToRest(Rule app) {
-        return toRest.apply(app);
-    }
+    public RestRule appToRest(Rule app, User user) {
+        RestRule restRule = toRest.apply(app);
+
+        //If the RestRule is null, toRest failed somehow, we need to disable the rule.
+        if(restRule == null){
+            try{
+                APILocator.getRulesAPI().disableRule(app, user);
+            } catch (DotDataException|DotSecurityException e){
+                Logger.error(this, "Error trying to disable Rule: " + app.getId(), e);
+            }
+        }
 
-    public Function<Rule, RestRule> appToRestFn() {
-        return toRest;
+        return restRule;
     }
 
     private final Function<Rule, RestRule> toRest = (app) -> {
-        Map<String, RestConditionGroup> groups = app.getGroups()
-                                                    .stream()
-                                                    .collect(Collectors.toMap(ConditionGroup::getId, groupTransform.appToRestFn()));
-
-        Map<String, Boolean> ruleActions = app.getRuleActions()
-                                                         .stream()
-                                                         .collect(Collectors.toMap(RuleAction::getId, ruleAction -> true));
-
-        return new RestRule.Builder()
-                .key(app.getId())
-                .name(app.getName())
-                .fireOn(app.getFireOn().toString())
-                .shortCircuit(app.isShortCircuit())
-                .priority(app.getPriority())
-                .enabled(app.isEnabled())
-                .conditionGroups(groups)
-                .ruleActions(ruleActions)
-                .build();
+        RestRule restRule = null;
+
+        try{
+            Map<String, RestConditionGroup> groups = app.getGroups()
+                    .stream()
+                    .collect(Collectors.toMap(ConditionGroup::getId, groupTransform.appToRestFn()));
+
+            Map<String, Boolean> ruleActions = app.getRuleActions()
+                    .stream()
+                    .collect(Collectors.toMap(RuleAction::getId, ruleAction -> true));
+
+            restRule = new RestRule.Builder()
+                    .key(app.getId())
+                    .name(app.getName())
+                    .fireOn(app.getFireOn().toString())
+                    .shortCircuit(app.isShortCircuit())
+                    .priority(app.getPriority())
+                    .enabled(app.isEnabled())
+                    .conditionGroups(groups)
+                    .ruleActions(ruleActions)
+                    .build();
+
+        } catch (Exception e) {
+            String ruleName = UtilMethods.isSet(app.getName()) ? app.getName() : "N/A";
+            Logger.error(this, "Error parsing Rule named: " + ruleName + " to ReST", e);
```

Comment:
throw new InternalServerException(.....). Or possibly throw a new custom exception type that extends that. RuleTransformException or similar.